internal/store/engine_state.go

```diff
@@ -106,7 +106,10 @@ type EngineState struct {
 
 	TelemetrySettings model.TelemetrySettings
 
-	UserConfigState model.UserConfigState
```

Comment:
ahhhh i'm glad you're excited for this change! Unfortunately, there are some tools that still read this off the engine state (most notably `tilty`), so we can't remove it yet. will send you a reference about it privately. but i think the other parts of this change are ok